File: objectiveai-sdk-rs/src/swarm/http.rs
//! HTTP client functions for Swarm endpoints.

use crate::{HttpClient, HttpError};

/// Lists all Swarms that have been used.
pub async fn list_swarms(
    client: &HttpClient,
    params: super::request::ListSwarmsRequest,
) -> Result<super::response::ListSwarmResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "swarms/list", Some(params))
        .await
}

/// Retrieves a specific Swarm.
pub async fn get_swarm(
    client: &HttpClient,
    params: super::request::GetSwarmRequest,
) -> Result<super::response::GetSwarmResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "swarms", Some(params))
        .await
}

/// Retrieves usage statistics for a specific Swarm.
pub async fn get_swarm_usage(
    client: &HttpClient,
    params: super::request::GetSwarmRequest,
) -> Result<super::response::UsageSwarmResponse, HttpError> {
    client
        .send_unary(reqwest::Method::POST, "swarms/usage", Some(params))
        .await
}

File: objectiveai-sdk-py/objectiveai_sdk/swarm/http.py
"""HTTP functions for swarm endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.swarm import (
        GetSwarmRequest,
        GetSwarmResponse,
        ListSwarmResponse,
        ListSwarmsRequest,
        UsageSwarmResponse,
    )


async def list_swarms(
    client: ObjectiveAI, params: ListSwarmsRequest,
) -> ListSwarmResponse:
    """List all swarms that have been used."""
    return await client.post_unary("swarms/list", params)


async def get_swarm(
    client: ObjectiveAI, params: GetSwarmRequest,
) -> GetSwarmResponse:
    """Retrieve a specific swarm."""
    return await client.post_unary("swarms", params)


async def get_swarm_usage(
    client: ObjectiveAI, params: GetSwarmRequest,
) -> UsageSwarmResponse:
    """Retrieve usage statistics for a specific swarm."""
    return await client.post_unary("swarms/usage", params)
